src/bot/storage/actual.ts

```diff
@@ -0,0 +1,251 @@
+          actualApi
+            .importTransactions(actualAccountId, transactions)
+            .catch((error) => {
+              logger(
+                `Error importing transactions for account "${accountName}": ${error.message}`,
+              );
+              return {
+                errors: [error.message],
+                added: [],
+                updated: [],
+              };
+            }),
+          onProgress(`Sending transactions for account "${accountName}"`),
+        ]);
+
+        if (importResponse.errors?.length) {
+          logger(
+            `Errors importing transactions: ${JSON.stringify(importResponse.errors)}`,
+          );
+          continue;
+        }
+
+        logger(
+          `Imported ${importResponse.added?.length || 0} transactions for account "${accountName}"`,
+        );
+        stats.added += importResponse.added?.length || 0;
+        stats.existing += importResponse.updated?.length || 0;
+      }
+
+      logger("transactions sent to Actual successfully!");
+
+      if (TRANSACTION_HASH_TYPE !== "moneyman") {
+        logger("Warning: TRANSACTION_HASH_TYPE should be set to 'moneyman'");
+      }
+    } catch (error) {
+      throw new Error(
+        `Failed to send transactions to Actual: ${error instanceof Error ? error.message : error}`,
+      );
+    }
+  }
+
+  private async init() {
+    if (this.initialized) return;
+
+    logger("init");
+
+    try {
+      const tempDir = path.join(os.tmpdir(), "moneyman-actual-data");
+      const dirExists = await fs.stat(tempDir).catch(() => false);
+      if (!dirExists) {
+        await fs.mkdir(tempDir, { recursive: true });
+      }
+
+      await actualApi.init({
+        dataDir: tempDir,
+        serverURL: ACTUAL_SERVER_URL,
+        password: ACTUAL_PASSWORD,
+      });
+
+      await actualApi.downloadBudget(ACTUAL_BUDGET_ID);
+
+      const actualAccounts = await actualApi.getAccounts();
+      const validActualAccountIds = new Set(actualAccounts.map((a) => a.id));
+      this.accountIdToNameMap = new Map(
+        actualAccounts.map((a) => [a.id, a.name]),
+      );
+
+      this.bankToActualAccountMap = this.parseActualAccounts(ACTUAL_ACCOUNTS!);
+
+      for (const [
+        bankAccountId,
+        actualAccountId,
+      ] of this.bankToActualAccountMap.entries()) {
+        if (!validActualAccountIds.has(actualAccountId)) {
+          const accountName =
+            this.accountIdToNameMap.get(actualAccountId) || actualAccountId;
+          logger(
+            `Warning: Actual Budget account "${accountName}" for bank account ${bankAccountId} does not exist`,
+          );
+          this.bankToActualAccountMap.delete(bankAccountId);
+        }
+      }
+
+      if (this.bankToActualAccountMap.size === 0) {
+        throw new Error(
+          "No valid account mappings found. Please check ACTUAL_ACCOUNTS configuration.",
+        );
+      }
+
+      this.initialized = true;
```

Comment:
Do you expect `saveTransactions` to be called more than once in the same process? IMHO there is no need to track the `initialized` state.

If the reason to check is the `accountIdToNameMap` and `bankToActualAccountMap` maps, you can use the size of them to early return from init, but as a wrote - `saveTransactions` is called once per storage